dali/operators/generic/reduce/reduce_with_mean_input.h

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_REDUCE_REDUCE_WITH_MEAN_INPUT_H__
+#define DALI_OPERATORS_GENERIC_REDUCE_REDUCE_WITH_MEAN_INPUT_H__
+
+#include <vector>
+#include <algorithm>
+
+#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/reduce/reductions.h"
+#include "dali/kernels/reduce/reduce_cpu.h"
+#include "dali/kernels/reduce/reduce_gpu.h"
+#include "dali/kernels/reduce/reduce_setup_utils.h"
+
+#define REDUCE_WITH_MEAN_TYPES_MAP ( \
+    ((uint8_t), (uint8_t, float)), \
+    ((int8_t), (int8_t, float)), \
+    ((uint16_t), (uint16_t, float)), \
+    ((int16_t), (int16_t, float)), \
+    ((uint32_t), (uint32_t, float)), \
+    ((int32_t), (int32_t, float)), \
+    ((uint64_t), (uint64_t, float)), \
+    ((int64_t), (int64_t, float)), \
+    ((float), (float)))
```

Comment:
Likewise - the output types for Standard Deviation and for Variance would differ (variance needs a 2x larger output type).